apps/user-portal/src/pages/user-profile.tsx

```diff
@@ -154,91 +179,140 @@ class UserProfilePageComponent extends React.Component<IComponentProps, any> {
-                        </Button>
-                        <Form.Field label={value}/></>
+                        <Form.Field>
+                            <label>{t("views:userProfile.inputFields.email")}</label>
+                        <label>{value}
+                            <Icon size="small" color="grey" id="emailEdit"
+                                  onClick={this.handleEdit} style={{marginLeft: "10px"}}
+                                  name="pencil alternate"/>
+                        </label>
+                        </Form.Field>
+                    </>
                 );
             }
         };
 
-        const handleEmailChange = (value) => {
-            if (this.state.emailEdit) {
+        const handlePInfoChange = () => {
+            if (this.state.pInfoEdit) {
                 return (<>
-                        <Segment padded>
-                            {t("views:userProfile.inputFields.email")}
-                            <Form.Input id="email" value={value} onChange={this.handleFieldChange}/>
-                            <div className="ui two buttons">
-                                <Button id="email" positive onClick={this.handleSave}>
+                        <Container>
+                            <Segment padded style={{width: "550px"}}>
+                                <Grid>
+                                    <Grid.Row columns={2}>
+                                        <Grid.Column >
+                                            <Form.Field>
+                                                <label>{t("views:userProfile.inputFields.organisation")}</label>
+                                                <input required value={this.state.organisation} id="organisation"
+                                                       onChange={this.handleFieldChange}/>
+                                            </Form.Field>
+                                        </Grid.Column>
+                                        <Grid.Column >
+                                            <Form.Field>
+                                                <label>{t("views:userProfile.inputFields.mobile")}</label>
+                                                <input required value={this.state.mobile} id="mobile"
+                                                       onChange={this.handleFieldChange}/>
+                                            </Form.Field>
+                                        </Grid.Column>
+                                    </Grid.Row>
+                                </Grid>
+                                <Divider hidden/>
+                                <Button id="pInfo" primary onClick={this.handleSave}>
                                     {t("common:save")}
                                 </Button>
-                                <Button id="emailEdit" onClick={this.handleCancel}>
+                                <Button id="pInfoEdit" secondary onClick={this.handleCancel}>
```

Comment:
Why do we need a `id` here?

Reply:
This id is passed into the event handler of the cancel button.